test/get-changelog-test.js

```diff
@@ -371,4 +371,63 @@ describe(getChangelog, function() {
     expect(changelog).to.include('[1.0.1]');
     expect(changelog).to.not.include('[1.0.0]');
   });
+
+  it('can generate in reverse order (revert)', async function() {
+    fixturify.writeSync(tmpPath, {
+      'packages': {
+        'my-app': {
+          'package.json': stringifyJson({
+            'name': '@scope/my-app',
+            'version': '1.0.0',
+          }),
+        },
+      },
+      'package.json': stringifyJson({
+        'private': true,
+        'workspaces': [
+          'packages/*',
+        ],
+      }),
+    });
+
+    await execa('git', ['add', '.'], { cwd: tmpPath });
+    await execa('git', ['commit', '-m', 'fix: old-release'], { cwd: tmpPath });
+
+    process.chdir(path.join(tmpPath, 'packages/my-app'));
+
+    await standardVersion({
+      path: path.join(tmpPath, 'packages/my-app'),
+      tagPrefix: '@scope/my-app@',
+      firstRelease: true,
+    });
+
+    let oldCommit = await getCurrentCommit(tmpPath);
+
+    fixturify.writeSync(tmpPath, {
+      'packages': {
+        'my-app': {
+          'index.js': 'foo',
+        },
+      },
+    });
+
+    await execa('git', ['add', '.'], { cwd: tmpPath });
+    await execa('git', ['commit', '-m', 'fix: foo'], { cwd: tmpPath });
+
+    await standardVersion({
+      path: path.join(tmpPath, 'packages/my-app'),
+      tagPrefix: '@scope/my-app@',
+    });
+
+    let newCommit = await getCurrentCommit(tmpPath);
+
+    await execa('git', ['reset', '--hard', oldCommit], { cwd: tmpPath });
+
+    let changelog = await getChangelog({
+      cwd: path.join(tmpPath, 'packages/my-app'),
+      fromCommit: newCommit,
+    });
+
+    expect(changelog).to.include('[1.0.0]');
```

Comment:
there is no other interaction with the changelog in this test -- I don't understand how it tests that the revert was successful

Reply:
Since the latest version is 1.0.1, you would expect a failure case to be version 1.0.1, or even 1.0.2. 1.0.0 indicates it went back in time. Keep in mind, this is a changelog section, not the result of an append, so in a normal changelog request, you would not have 1.0.0 in it. (perhaps the call should have been called `getChangelogSection`).

Reply:
I updated it to

```
    expect(changelog).to.include('[1.0.0]');
    expect(changelog).to.not.include('[1.0.1]');
```

to help.